### PR TITLE
feat: added signup and login page with auth

### PR DESCRIPTION
**Pull request by Developers-codz**
## Files To Check

`src/hooks/auth.js`
`src/pages/login/Login.jsx`
`src/pages/signup/Signup.jsx`

## File Not To Check
Please ignore other files.

Added signup and login pages and did API call for login and signup functionalities and storing the encodedToken to localstorage. Note: Private route is not added yet

### [live link](https://deploy-preview-6--distracted-payne-21e589.netlify.app/)

### CODE REVIEW

File: src/pages/signup/Signup.jsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+              id="sign-pass"
+              className="input-field"
+              placeholder="Enter your password here"
+              onChange={inputHandler}
+              required
+            />{" "}
+            <i
+              className={classNames(" eye reset", {
+                "fa fa-eye": type === "text",
+                "far fa-eye-slash": type === "password",
```
Comment (anik31) [inline src/pages/signup/Signup.jsx:68]: appreciation - learnt a new thing about classnames

**Review by anik31 — APPROVED** [review]
Good work, code is clean & readable.

File: src/pages/signup/Signup.jsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+              })}
+              onClick={() =>
+                setType((type) => (type === "password" ? "text" : "password"))
+              }
```
Comment (ApurvaSawant11) [inline src/pages/signup/Signup.jsx:72]: Liked your approach. 
Just one more thing. Hide the inbuilt `eye` of the input. Both the eyes are visible.

File: src/hooks/auth.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+const signupHandler = async (e, formData, setFormData, formObj) => {
+  e.preventDefault();
+  try {
+    const response = await axios.post(
+      "/api/auth/signup",
+      JSON.stringify(formData)
+    );
+    // saving the encodedToken in the localStorage
+    localStorage.setItem("token", response.data.encodedToken);
+    console.log(response.data);
```
Comment (ApurvaSawant11) [inline src/hooks/auth.js:34]: Remove all `console.log` 's if not needed

File: src/pages/login/login.css [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  font-size: 1rem;
+}
+.login-btn:hover {
+  transform: scale(1.03, 1.03);
+  font-weight: 500;
+}
+.signin-section {
+  margin-top: 1rem;
+  margin-bottom: 1rem;
+}
```
Comment (ApurvaSawant11) [inline src/pages/login/login.css:63]: About `login.css` and `signup.css`.

One thing I noticed is that 95% of CSS is same. You can create a common CSS file and use comma(,) separator to assign those properties to both login and signup.

**Review by ApurvaSawant11 — APPROVED** [review]
The code looks well structured and easier to read.

Just gave a suggestion.

File: src/hooks/auth.js (hunk 3 above)
Comment (Developers-codz) [reply src/hooks/auth.js:34]: As of now its needed until I don't complete all auth part

File: src/pages/login/login.css (hunk 4 above)
Comment (Developers-codz) [reply src/pages/login/login.css:63]: Will look into that later after completing the basic requirements, actually these is coming from previously build screens in html css

File: src/pages/signup/Signup.jsx (hunk 2 above)
Comment (Developers-codz) [reply src/pages/signup/Signup.jsx:72]: that eye is CSS class used that for positioning the `eye` icon

**Pull request merged**